### PR TITLE
✨ Add script for making HPO CodeSystem

### PR DESCRIPTION
**Pull request by fiendish**
Part of https://github.com/d3b-center/clinical-data-flow/issues/36

### CODE REVIEW

File: scripts/hpo_codesystem.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env python
+
```
Comment (liberaliscomputing) [inline scripts/hpo_codesystem.py]: `import argparse`

File: scripts/hpo_codesystem.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import json
+import os
+import sys
+
+# OWL to FHIR transformer compiled from https://github.com/aehrc/fhir-owl
+FHIR_OWL_BIN = "fhir-owl-1.1.0-SNAPSHOT.jar"
+
```
Comment (liberaliscomputing) [inline scripts/hpo_codesystem.py]:
```
parser = argparse.ArgumentParser()
parser.add_argument(
    'terminology',
    help=(
        '''
        Terminology short name for fetching the OWL file from OBO PURLs
        (e.g. hpo, mondo, ncit)
        '''
    )
)
```

File: scripts/hpo_codesystem.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+# Convert OWL file to FHIR CodeSystem
+ret = os.system(f'wget -N http://purl.obolibrary.org/obo/hp.owl && java -jar {FHIR_OWL_BIN} -i hp.owl -o hpo_codesystem.json -id hpo -name "Human Phenotype Ontology" -content not-present -mainNs http://purl.obolibrary.org/obo/HP_ -descriptionProp http://purl.org/dc/elements/1.1/subject -status active -codeReplace _,:')
```
Comment (liberaliscomputing) [inline scripts/hpo_codesystem.py]:
```
ret = os.system(
    f'''
    wget -N http://purl.obolibrary.org/obo/{terminology}.owl && 
    java -jar {FHIR_OWL_BIN} -i {terminology}.owl -o codesystem_{terminology}.json -codeReplace _,:
    '''
)
```

File: scripts/hpo_codesystem.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+with open("hpo_codesystem.json") as hj:
+    hpo = json.load(hj)
+
+hpo["concept"] = [c for c in hpo["concept"] if c["code"].startswith("HP:")]
+for c in hpo["concept"]:
+    c.pop("designation", None)
+
+with open("hpo_codesystem.json", "w") as hj:
+    json.dump(hpo, hj)
```
Comment (liberaliscomputing) [inline scripts/hpo_codesystem.py]:
```
# Remove cross-references to other ontologies
with open(f'codesystem_{terminology}.json') as f:
    codesystem = json.load(f)

codesystem['concept'] = [
    concept for concept in codesystem['concept']
    if concept['code'].startswith(f'{terminology.upper()}:')
]
for concept in codesystem['concept']:
    concept.pop('designation', None)
codesystem['count'] = len(codesystem['concept'])

with open(f'codesystem_{terminology}.json', 'w') as f:
    json.dump(codesystem, f)
```

**Comment by liberaliscomputing** [comment]
I made some local tweaks so that it can run more generically :) Tested with mondo and ncit! Thanks tons, @fiendish!

File: scripts/hpo_codesystem.py (hunk 3 above)
Comment (fiendish) [reply scripts/hpo_codesystem.py]: This modification would fail if you type hpo instead of hp. I think we need to provide a few safe aliases.

Comment (liberaliscomputing) [reply scripts/hpo_codesystem.py]: Oh, it was hp instead of hpo! Totally agreed!

**Comment by znatty22** [comment]
We can make this script more generic in the future. Good enough to merge for now.

**Pull request merged**